src/app.jsx

```diff
@@ -606,23 +611,31 @@ const NavigatorCardBody = ({
         );
     };
 
-    if (isGrid) {
+    if (loadingFiles)
         return (
-            <CardBody onClick={resetSelected} id="navigator-card-body">
-                <Flex id="folder-view">
-                    {sortedFiles.map(file => <Item file={file} key={file.name} />)}
-                </Flex>
-            </CardBody>
-        );
-    } else {
-        return (
-            <ListingTable
-              id="folder-view"
-              className="pf-m-no-border-rows"
-              variant="compact"
-              columns={[_("Name")]}
-              rows={sortedFiles.map(file => ({ columns: [{ title: <Item file={file} key={file.name} /> }] }))}
-            />
+            <Flex justifyContent={{ default: "justifyContentCenter" }}>
+                <Spinner />
+            </Flex>
         );
+    else {
```

Comment:
To avoid the nesting (which makes code harder to read) you can just omit the 'else' part completely.

The return will anyhow end the call.